enterprise/api/v1/views.py

```diff
@@ -450,6 +454,40 @@ def toggle_universal_link(self, request, pk=None):
         headers = self.get_success_headers(response_body)
         return Response(response_body, status=HTTP_200_OK, headers=headers)
 
+    @action(methods=['post'], detail=True, permission_classes=[permissions.IsAuthenticated])
+    @permission_required('enterprise.can_access_admin_dashboard', fn=lambda request, pk: pk)
+    def unlink_users(self, request, pk=None):  # pylint: disable=unused-argument
+        """
+        Unlinks users with the given emails from the enterprise.
+        """
+
+        serializer = serializers.EnterpriseCustomerUnlinkUsersSerializer(
+            data=request.data
+        )
+
+        serializer.is_valid(raise_exception=True)
+
+        enterprise_customer = self.get_object()
+        emails_to_unlink = serializer.data.get('emails', [])
+        is_relinkable = serializer.data.get('is_relinkable', True)
+
+        with transaction.atomic():
+            for email in emails_to_unlink:
+                try:
+                    models.EnterpriseCustomerUser.objects.unlink_user(
+                        enterprise_customer=enterprise_customer,
+                        user_email=email,
+                        is_relinkable=is_relinkable
+                    )
+                except (models.EnterpriseCustomerUser.DoesNotExist, models.PendingEnterpriseCustomerUser.DoesNotExist):
+                    msg = "User with email {} does not exist in enterprise {}.".format(email, enterprise_customer)
+                    LOGGER.warning(msg)
+                except Exception as exc:
+                    LOGGER.exception(exc)
+                    raise
```

Comment:
Nit - you could do something like this to make the logging a little more helpful, here:
```python
message = f'Could not unlink {email} from {enterprise_customer}'
raise EnterpriseCustomerUserUnlinkError(message) from exc. # you have to define this Exception class somewhere above
```